packages/react-composites/tests/browser/common/chatTestHelpers.ts

```diff
@@ -35,6 +35,28 @@ export const chatTestSetup = async ({
   await Promise.all(pageLoadPromises);
 };
 
+export const chatTestSetupWithPerUserArgs = async ({
+  pages,
+  usersWithArgs,
+  serverUrl
+}: {
+  pages: Page[];
+  usersWithArgs: { user: ChatUserType; qArgs?: { [key: string]: string } }[];
+  serverUrl: string;
+}): Promise<void> => {
+  const newUsers = await createChatThreadAndUsers(TEST_PARTICIPANTS);
```

Comment:
This function repeats the bug on line 24: it accepts `users` but then creates new users ignoring the ones passed in. That is very surprising, and we lost a few hours last week debugging it.

We should instead fix this bug both here and on line 24. If the function accepts `users`, it must respect those users. There is no need to recreate users and thread, just reload the page for the given users with the args you want.

Reply:
The reason the thread is recreated is to ensure each test is using it's own thread. Ideally we should remove the fixture code that initially sets up a thread as this is only run before all tests (not before every test which is what we need)

Reply:
(But agree that it is bad we pass in users and then recreate new ones)